Command "migration:show" should not return a non-zero exit code when there are unapplied migrations

## Issue Description

The `migration:show` command is returning a non-zero exit code when there are unapplied migrations, but this behaviour makes no sense, especially outside of a CI environment.

It should return a 0 exit code by default, and a CLI argument could be used to force a non-zero code.

### Expected Behavior

Clean output with no errors (i.e. the exit code should be 0 by default).

### Actual Behavior

If I simply wish to view a list of migrations, I get the following output:

```
> @x/api@1.0.0 typeorm D:\Programming\Projects\x\v4\api
> ts-node ./node_modules/typeorm/cli.js "migration:show"

query: SELECT * FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA` = 'x' AND `TABLE_NAME` = 'migrations'
query: SELECT * FROM `x`.`migrations` `migrations`  ORDER BY `id` DESC
 [ ] testIng1612699019592
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @x/api@1.0.0 typeorm: `ts-node ./node_modules/typeorm/cli.js "migration:show"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @x/api@1.0.0 typeorm script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Glen\AppData\Roaming\npm-cache\_logs\2021-02-07T12_13_13_421Z-debug.log
```

This is a mess. It's difficult to even see the list of migrations.

### Steps to Reproduce

1. Create a migration.
2. Run the `migration:show` command.

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    | Windows 10         |
| Node.js version     | v14.15.4 | <!-- run `node -v` to obtain this -->
| Typescript version  | v4.1.3 | <!-- run `npm list typescript` to obtain this -->
| TypeORM version     | v0.2.30 | <!-- run `npm list typeorm` to obtain this -->

### Additional Context

https://github.com/typeorm/typeorm/pull/4173#commitcomment-36622884
https://github.com/typeorm/typeorm/pull/4173#issuecomment-770399199
https://github.com/typeorm/typeorm/pull/4173#issuecomment-770472818

### Relevant Database Driver(s)

- [x] `aurora-data-api`
- [x] `aurora-data-api-pg`
- [x] `better-sqlite3`
- [x] `cockroachdb`
- [x] `cordova`
- [x] `expo`
- [x] `mongodb`
- [x] `mysql`
- [x] `nativescript`
- [x] `oracle`
- [x] `postgres`
- [x] `react-native`
- [x] `sap`
- [x] `sqlite`
- [x] `sqlite-abstract`
- [x] `sqljs`
- [x] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [ ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [x] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

/cc @Juddling

--- a/src/commands/MigrationShowCommand.ts
+++ b/src/commands/MigrationShowCommand.ts
@@ -24,6 +24,9 @@ export class MigrationShowCommand implements yargs.CommandModule {
         alias: "f",
         default: "ormconfig",
         describe: "Name of the file with connection configuration."
+      })
+      .option("quiet", {
+        describe: "Silences the error exit code if there are unapplied migrations"
       });
   }
 
@@ -46,8 +49,9 @@ export class MigrationShowCommand implements yargs.CommandModule {
       const unappliedMigrations = await connection.showMigrations();
       await connection.close();
 
-      // return error code if there are unapplied migrations for CI
-      process.exit(unappliedMigrations ? 1 : 0);
+      const silenceError = args.quiet !== undefined ? true : false;
+      // return error code if there are unapplied migrations
+      process.exit(unappliedMigrations && !silenceError ? 1 : 0);
 
     } catch (err) {
       if (connection) await (connection as Connection).close();

--- a/src/commands/MigrationShowCommand.ts
+++ b/src/commands/MigrationShowCommand.ts
@@ -24,9 +24,6 @@ export class MigrationShowCommand implements yargs.CommandModule {
         alias: "f",
         default: "ormconfig",
         describe: "Name of the file with connection configuration."
-      })
-      .option("quiet", {
-        describe: "Silences the error exit code if there are unapplied migrations"
       });
   }
 
@@ -49,9 +46,7 @@ export class MigrationShowCommand implements yargs.CommandModule {
       const unappliedMigrations = await connection.showMigrations();
       await connection.close();
 
-      const silenceError = args.quiet !== undefined ? true : false;
-      // return error code if there are unapplied migrations
-      process.exit(unappliedMigrations && !silenceError ? 1 : 0);
+      process.exit(0);
 
     } catch (err) {
       if (connection) await (connection as Connection).close();

--- a/src/commands/MigrationShowCommand.ts
+++ b/src/commands/MigrationShowCommand.ts
@@ -43,7 +43,7 @@ export class MigrationShowCommand implements yargs.CommandModule {
         logging: ["query", "error", "schema"]
       });
       connection = await createConnection(connectionOptions);
-      const unappliedMigrations = await connection.showMigrations();
+      await connection.showMigrations();
       await connection.close();
 
       process.exit(0);

